installer/scripts/omsadmin.sh

```diff
@@ -469,6 +469,39 @@ onboard()
             fi
         fi
     fi
+
+    #check for workspaces alredy onboarded
+
+    local found_ws=0
+    local ws_conf_dir=$ETC_DIR/conf
+
+    if [ -h ${ws_conf_dir} ]; then
+        local primary_ws_id=''
+        if [ -f ${ws_conf_dir}/omsadmin.conf ]; then
+            primary_ws_id=`grep WORKSPACE_ID ${ws_conf_dir}/omsadmin.conf | cut -d= -f2`
+        fi
+
+        if [ "${primary_ws_id}" != "" ]; then
+            found_ws=1
+        else
+            for ws_id in `ls -1 $ETC_DIR | grep -E '^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$'`
+            do
+                found_ws=1
+            done
+        fi
+    else
+        # no default conf folder, check all the potential workspace folders
+        for ws_id in `ls -1 $ETC_DIR | grep -E '^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$'`
+        do
+            found_ws=1
+        done
+    fi
+
+    if [ $found_ws -ne 0 ]; then
+        echo "Already Onboarded to a workspace, please un-onboard first before trying to onboard to a new workspace"
+        return 0
```

Comment:
why 0? it should return a code that is sent back to the python code and ultimately should show up in the VM extensoin tab in portal.

Reply:
I think this case would only arise if the customer had installed bundle, stopped the agent, and then tried to enable the extension. But I agree, probably should not return 0

Reply:
Yeah this case only happens when the agent is installed successfully and then is stopped manually and then cx tries to re-onboard. The only caveat is that if the agent is installed as a VM extension then Waagent will repeatedly call enable on it until it succeeds, so i guess yeah it is a good idea to use an error code instead of 0. Should I use 53 (The extension failed due to missing or wrong configuration parameters) instead?

Reply:
of the 50s, 53 is probably best but IDK if it is exactly right. regardless, it should fall into one of the codes we ignore for SLA purposes

Reply:
Hmm thinking again on this, shouldnt it still be 0? This case isnt really an enable failure. It is simply preventing re-onboaring step but the enable process will still continue after this and it will restart the stopped omsagent in case the agent was onboarded before but is manually stopped. Which is what we expect from the Enable command. If we return 53 then the Enable will fail. Not sure which should be the right behavior here. Any ideas @simathih @hestolz ?

Reply:
What if the cx is trying to onboard a different workspace. Will that be caught by either the bundle onboard script or the extension? I know we have stop on multiple connections for the extension now, but the same logic is not part of bundle

Reply:
It will first be caught by the extension code, and then in the bundle itself as well. Currently the Service Control is_running method returns a non 0 exit code for a diff workspace. This wasn't by design but is rather a happy co-incidence (issue) as it prevents multihoming by default. I had added a small change to fix that issue as part of my multihoming work but since that is on hold for now, I decided to leave the service_control script as it is to avoid unnecessary MH problems.

Reply:
if this returns 0, then extension will assume install was successful and skip?

Reply:
The issue is unlikely to happen in case of install as the agent will not be onboarded to another workspace before and it won't have the workspace-id directories. But yeah i agree that there can be edge cases or cases with incomplete installation or uninstallation that can hit this issue. So, after discussion with Henry, **I changed the exit code to 53** as that seemed to be the best option.